### ISSUE TITLE
HMR not working in Storybook with Vanilla Extract

### ISSUE THREAD

**Issue by hanshank** [vanilla-extract-css/vanilla-extract#905]
### Describe the bug

**Run repro repo locally and then:**
1. While Storybook is running in the browser (Chrome), navigate to src/stories/button.css.ts
2. Make a change to the styles, e.g. change fontSize property to '100px'.

**What happens**
HMR kicks in but Storybook can no longer find the component. I have to do a full page reload to get the updated component

**What should happen**
HMR kicks in and the component styles gets updated without any issues.

I don't know what's going on, but I did follow the install guide in your wiki here: https://vanilla-extract.style/documentation/integrations/webpack/ 

After HMR in Storybook kicks in I'm getting the following:

### Reproduction

https://github.com/hanshank/storybook-and-vanilla-extract

### System Info

```shell
System:
    OS: macOS 12.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 80.75 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Firefox: 106.0.1
    Safari: 16.0
    Safari Technology Preview: 16.4
  npmPackages:
    @vanilla-extract/css: ^1.9.1 => 1.9.1 
    @vanilla-extract/webpack-plugin: ^2.2.0 => 2.2.0 
    webpack: ^5.74.0 => 5.74.0
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ybot1122** [comment]
Hello; Since Storybook is using `splitChunks` optimization: https://github.com/storybookjs/storybook/blob/next/code/lib/builder-webpack5/src/preview/iframe-webpack.config.ts#L282-L284 This is not the default value for webpack (https://webpack.js.org/plugins/split-chunks-plugin/)

All you have to do in your `.storybook/main.js` is to `delete config.optimization.splitChunks` and should be fixed.

**Comment by hanshank** [comment]
@ybot1122 That's a great find! It does fix the HMR issue for me!!! 

I would rather not delete the splitChunks optimization, but I found another approach in this issue: https://github.com/vanilla-extract-css/vanilla-extract/issues/746#issuecomment-1207390899 - I modified @danielberndt's approach ever so slightly to maintain all the default properties of the splitChunks object. But it basically his solution I am using, so thank you!

TLDR: 

1. Copy the official default splitChunks object from Webpack: https://webpack.js.org/plugins/split-chunks-plugin/#optimizationsplitchunks. Wish there was a way to extend the default Webpack config instead, but according to [this](https://github.com/webpack/webpack.js.org/issues/5174) comment that's not an option 😢
2. In `.storybook/main/.js` add this to `webpackFinal`:

```js
module.exports = {
  // ... other config properties
  webpackFinal: async (config) => {
     // vanilla extract setup

    // add this
    if(process.env.NODE_ENV === 'development') {
      config.optimization = {
        ...config.optimization,
          splitChunks: {
             chunks: 'async',
             minSize: 20000,
             minRemainingSize: 0,
             minChunks: 1,
             maxAsyncRequests: 30,
             maxInitialRequests: 30,
             enforceSizeThreshold: 50000,
             cacheGroups: {
               defaultVendors: {
                 test: `[\\/]node_modules[\\/](?!.*vanilla-extract)`, // this is the line that fixes it
                 priority: -10,
                 reuseExistingChunk: true,
               },
              default: {
                minChunks: 2,
                priority: -20,
                reuseExistingChunk: true,
              },
          },   
        }
      }
    }
  }
}
```

The `cacheGroups.defaultVendors.test` property was changed from the default `/[\\/]node_modules[\\/]/` => `[\\/]node_modules[\\/](?!.*vanilla-extract)`

If someone has a better solution I'm all ears!

**Comment by JSerFeng** [comment]
For those of who comes here for solution.

The HMR broken reason is here: https://github.com/web-infra-dev/rsbuild/issues/6049#issuecomment-3248531696

The workaround is to enforce the chunk name to be consistent for dev.

So in dev, you can use create a `splitChunk.cacheGroups` to make sure all modules that vanilla extract plugin created in the same fixed name chunk.

Make sure the `name` config here is only for development, not for production, as in production, we should never merge async modules and initial modules in the same chunk

```js
module.exports = {
  optimization: {
    splitChunks: {
      cacheGroups: { 
         vanillaCss: {
            minSize: 0,
            chunks: 'all',
            test: /@vanilla-extract\/webpack-plugin/,
            priority: 1000,
            name: process.env.NODE_ENV === 'development' ? 'vanilla-extract' : undefined
          }
      }
    }
  }
}
```
